backend/presentation/services.py

```diff
@@ -0,0 +1,54 @@
+from typing import OrderedDict
+
+from presentation.models import Presentation, Tag
+
+
+class PresentationService:
+    def create_presentation(self, validated_data: OrderedDict) -> Presentation:
```

Comment:
Maybe you could abstract incoming data into separate object (`typing.NamedTuple` or using class with `@dataclass(frozen=True)` decorator)
```suggestion
    def create_presentation(self, presentation_data: PresentationData) -> Presentation:
```